Error ck 1to3

stephennfitzgerald opened the pull request:
Additional error checks moved into validate sub routine. Minor changes to the make_test_sam.pl script to allow for incorporation of read-tags in the the test BAM file headers.

stephennfitzgerald committed:
minor changes to allow for overloading of biotype field with gencode_basic, TSL and appris transcript attributes.:
eg.
old: protein_coding,protein_coding
new: protein_coding:gencode_basic:tsl1:appris_pi,protein_coding ...

stephennfitzgerald committed:
minor changes to allow for overloading of biotype field with gencode…basic, TSL and appris transcript attributes.
eg.
old: protein_coding,protein_coding
new: protein_coding:gencode_basic:tsl1:appris_pi,protein_coding ...

stephennfitzgerald committed:
some error checks on the yaml file to ensure sample names, tag and BAM file names have not been duplicated

stephennfitzgerald committed:
Merge branch 'master' of https://github.com/iansealy/DETCT into ian_origin

Conflicts:
	lib/DETCT/GeneFinder.pm
	lib/DETCT/Transcript.pm

stephennfitzgerald committed:
some error checks for the YAML and BAM files

stephennfitzgerald committed:
new test files added to check for errors in YAML file

stephennfitzgerald committed:
tests for errors in YAML files
duplicate sample names,
different sample names pointing to the same TAG and BAM files
BAM files missing an index file

stephennfitzgerald committed:
tests for errors in YAML file (duplicate sample names, missing BAM file index, different samples pointing to same tag and BAM)

iansealy commented:
Would it be possible to move the checks into the existing validate method please? That way the error checking will still happen even if the analysis is created via the new() method and then adding samples rather than just by calling new_from_yaml(). It'll mean that the checks will need to look at the actual Sample objects though, rather than the YAML object. Thanks.

stephennfitzgerald committed:
extra error checking for:
Duplicated sample name
The same tag and BAM file in several samples
Tag sequences present in the sample but missing from the BAM file
Missing BAM file index

stephennfitzgerald committed:
some extra error checking for:

Duplicated sample name
The same tag and BAM file in several samples
Tag sequences present in the sample but missing from the BAM file
Missing BAM file index

stephennfitzgerald committed:
extra error checking for:

Duplicated sample name
The same tag and BAM file in several samples
Tag sequences present in the sample but missing from the BAM file
Missing BAM file index

stephennfitzgerald committed:
added BAM file name to error missing tags check
